rpcs3/Emu/Io/usb_device.cpp

```diff
@@ -187,32 +198,67 @@ bool usb_device_emulated::open_device()
 	return true;
 }
 
-s32 usb_device_emulated::get_descriptor(u8 type, u8 index, u8* ptr, u32 /*max_size*/)
+void usb_device_emulated::get_descriptor(u8 type, u8 index, u8* buf, u32 buf_size)
 {
-	if (type == USB_DESCRIPTOR_STRING)
+	switch (type)
+	{
+	case USB_DESCRIPTOR_DEVICE:
+	{
+		buf[0] = device.bLength;
+		buf[1] = device.bDescriptorType;
+		memcpy(buf + 2, device.data, buf_size - 2);
```

Comment:
buf_size may be bigger than device.data.size

Reply:
I think buf_size can't be bigger than device.data.size().
Look at usb_device.h:124, the size of u8 data[0xFF] is fixed, and according to https://www.beyondlogic.org/usbnutshell/usb5.shtml the max reported size (bLength) is 255 (one byte).

Reply:
I still updated the code, however.